### ISSUE TITLE
getaddrinfo ENOTFOUND causes monitors to freeze

### ISSUE THREAD

**Issue by shalak** [louislam/uptime-kuma#4058]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### Description

`getaddrinfo ENOTFOUND` errors make the monitors stop working

### 👟 Reproduction steps

When my local DNS service is down, all my monitors turn red and show errors - as expected:
```
getaddrinfo ENOTFOUND myhost.mydomain.com
```
However, after the DNS is up, they never restart again, I'm stuck in "down". Even restarting the uptime-kuma docker container doesn't help. What helps is either 
- disabling and enabling monitor
- enabling manual maintenance mode which contains all services, waiting 1 cycle and disabling it

### 👀 Expected behavior

`getaddrinfo ENOTFOUND` errors do not stop the monitors

### 😓 Actual Behavior

`getaddrinfo ENOTFOUND` errors do stops the monitors

### 🐻 Uptime-Kuma Version

1.23.6

### 💻 Operating System and Arch

Ubuntu 22.04.3 LTS

### 🌐 Browser

Firefox 119.0 (64-bit)

### 🐋 Docker Version

Docker version 24.0.7, build afdd53b

**Comment by shalak** [comment]
I just noticed something, I do not know if it's relevant, but I have two alert services:
- REST API-based signal notification
- email-based notification

It appears that only the monitors that have signal alerting are affected. The monitor that has email notification (i.e. signal REST API itself in my case) seems to be re-starting itself...

**Comment by shalak** [comment]
Huh, it may not at all be related to `ENOTFOUND` issue, but to the fact that the Signal REST API has a timeout bug:

https://github.com/bbernhard/signal-cli-rest-api/issues/453

So - the monitor detects an issue, calls signal REST API to send a message, the message is received, however the POST itself times out, thus hanging the monitor.

**Comment by lyup** [comment]
> So - the monitor detects an issue, calls signal REST API to send a message, the message is received, however the POST itself times out, thus hanging the monitor.

Ran into this exactly today. Wasn't sure what was going on until I read this. I debugged a bit and found what was happening is

* `sendNotificatin` can sometimes timeout or never returns https://github.com/louislam/uptime-kuma/blob/master/server/model/monitor.js#L952 
* once that happened, it's possible that we don't have a `previousBeat` in the time window, a `push` monitor item will never recover https://github.com/louislam/uptime-kuma/blob/master/server/model/monitor.js#L634

**Comment by louislam** [comment]
> sendNotificatin can sometimes timeout or never returns https://github.com/louislam/uptime-kuma/blob/master/server/model/monitor.js#L952

Good point, I think we should not add `await` before `Monitor.sendNotification(isFirstBeat, this, bean);`.

**Comment by ertankucukoglu** [comment]
I also see `ENOTFOUND` error in status for a newly added monitor. This is HTTP(s) monitor and just checking a web site is accessible for HTTP status 200-299 range.
When I check using `nslookup` and the domain is OK at OS level. Console command finds the IP of the domain. However, system reports this as ENOTFOUND for more than 20 minutes. I have no web coding knowledge so I cannot track this down.
Since original title is related I am posting this in here.
Uptime Kuma Version: 2.2.1
